File: socketd-core/src/main/java/org/noear/socketd/broker/Broker.java
package org.noear.socketd.broker;

import org.noear.socketd.client.Client;
import org.noear.socketd.client.ClientConfig;
import org.noear.socketd.server.Server;
import org.noear.socketd.server.ServerConfig;

/**
 * 经纪人
 *
 * @author noear
 * @since 2.0
 */
public interface Broker<S extends ServerConfig, C extends ClientConfig> {
    static Broker getInstance() {
        return null;
    }

    Server createServer(S config);

    Client createClient(C config);
}

File: javascript/socketd-examples/demo02-node-ws/_broker.js

const {SocketD}  = require('@noear/socket.d');

async function main() {
    let server = SocketD.createServer("sd:ws")
        .config(c => c.port(8602).fragmentHandler(SocketD.newBrokerFragmentHandler()))
        .listen(SocketD.newBrokerListener())
        .start();


    await SocketD.createClient("sd:ws://127.0.0.1:8602?@=app1")
        .listen(SocketD.newEventListener().doOn("hello", (s, m) => {
            s.reply(m, SocketD.newEntity("me too!"));
        }))
        .openOrThow();

    let session2 = await SocketD.createClient("sd:ws://127.0.0.1:8602?@=app2")
        .openOrThow();

    let r = await session2.sendAndRequest("hello", SocketD.newEntity("hi").at("app1")).await();
    console.info(r.dataAsString());
}

main();